site/docs/v1/tech/installation-guide/docker.adoc

```diff
@@ -190,15 +190,27 @@ To upgrade FusionAuth when running with `docker-compose`:
 . Start it up: `docker-compose up`.
 . Login to the administrative UI. 
 
-If there were migrations required, they'll be applied automatically to your database if the installation link:/docs/v1/tech/installation-guide/fusionauth-app#runtime-modes[has a runtime mode of development]. If the installation is in production mode, make sure you apply the migrations out of band. 
+=== Migrations
+If there were migrations required, they'll be applied automatically to your database if the installation link:/docs/v1/tech/installation-guide/fusionauth-app#runtime-modes[has a runtime mode of development]. 
+
+==== Prior to 1.19
+
+If the installation is in `production` mode make sure you apply the migrations out of band. 
+
+==== Version 1.19 or later
+
+You can use `silent` mode for db upgrades in `development` or `production` runtime modes. The runtime modes are no longer used to identify if silent mode is desired, instead set the `fusionauth-app.silent-mode` configuration value to `true`. See the link:/docs/v1/tech/reference/configuration/[configation reference] or the link:/docs/v1/tech/guides/silent-mode/[silent mode guide] for more information.
+
+If you want to manage your own upgrades, performing the SQL migrations out of band is still a good option.
+
+=== Docker tags
```

Comment:
I think you're close...  

The logic is like this:

1. If runtime mode is development OR silent mode is enabled, we are eligible to enter maint. mode. 
   - development mode -> maint. mode eligible.
   - production mode + silent mode = maint. mode eligible.
   - Else, no maint. mode (silent or otherwise)
2. Now we are in maint. mode. 
   - Silent mode enabled? Yes, perform any upgrades in a non-interactive fashion. 
   - Silent mode not enabled (this means we must be development mode) use the interactive maint. mode and prompt the user to complete the db schema update. 

I think I would just document the current behavior and then call out the differences if you're running an older version (prior to 1.19.0).

Reply:
OK, I think I have it.